### ISSUE TITLE
requires_auth method

### ISSUE THREAD

**Issue by IsmailM** [phenopolis/phenopolis#90]
Can someone explain [this method](https://github.com/phenopolis/phenopolis/blob/master/views/__init__.py#L197) to me?

```python
def requires_auth(f):
    @wraps(f)
    def decorated(*args, **kwargs):

        if session:
          if 'user' in session: 
             return f(*args, **kwargs)
          else:
             print 'login'
             if config.LOCAL:
                 return redirect('login')
             else:
                 return redirect('https://uclex.cs.ucl.ac.uk/login')

        print 'method', request.method
        error=None
        if request.method == 'POST':
          username=request.form['username']
          password=request.form['password']
          if check_auth(username,password):
             return f(*args, **kwargs)
          else:
             if config.LOCAL:
                 return render_template('login.html', error='Invalid Credentials. Please try again.')
             else:
                 return redirect('https://uclex.cs.ucl.ac.uk/login')
    return decorated

```

What happens if there is no session and the request is not a 'POST'? (i.e. a GET with no session)?

imo, there should be an `else` after the POST if and simply return a 401

**Comment by pontikos** [comment]
Hi Ismail yes I agree.  There should be an else and return 401.

**Comment by IsmailM** [comment]
Fixed in #92
